### PR TITLE
[W11H03] fix testLoginFailed() and testLoginGarbage()

### LINKED ISSUE #279: login - unsuccessful test incorrect
The test expects username and password fields to be null after failure to login despite the specification stating that these fields are supposed to get set as soon as the token request succeeds.

### LINKED ISSUE #280: login - garbage data test incorrect
See #279.

### PR DESCRIPTION
**Pull request by felixhauptmann**
* fix #279 
* fix #280

### CODE REVIEW

File: w11h03/test/pgdp/networking/HTTPTest.java [hunk 1]
```diff
@@ -146,8 +146,8 @@ void testLoginFailed() {
                 })
                 .respond(422, error);
         assertFalse(dataHandler.login("Alfred", "Wagner"));
-        assertNull(getField(dataHandler, "username"));
-        assertNull(getField(dataHandler, "password"));
+        assertEquals("Alfred", getField(dataHandler, "username"));
+        assertEquals("Wagner", getField(dataHandler, "password"));
```
Comment (felixhauptmann) [inline w11h03/test/pgdp/networking/HTTPTest.java:150]:
```suggestion
```

File: w11h03/test/pgdp/networking/HTTPTest.java [hunk 2]
```diff
@@ -169,8 +169,8 @@ void testLoginGarbage() {
                 })
                 .respond(500, "Server's on fire");
         assertFalse(dataHandler.login("Alfred", "Wagner"));
-        assertNull(getField(dataHandler, "username"));
-        assertNull(getField(dataHandler, "password"));
+        assertEquals("Alfred", getField(dataHandler, "username"));
+        assertEquals("Wagner", getField(dataHandler, "password"));
```
Comment (felixhauptmann) [inline w11h03/test/pgdp/networking/HTTPTest.java:173]:
```suggestion
```

**Review by felixhauptmann — COMMENTED** [review]
@BjarneHa Would that be okay?

**Comment by nilsreichardt** [comment]
Is #282 resolving this already?

**Comment by BjarneHa** [comment]
Nope

**Comment by nilsreichardt** [comment]
I agree with @BjarneHa (https://github.com/MaximilianAnzinger/pgdp2223-tests/issues/279#issuecomment-1399319885) that we should keep the assertions because otherwise it can easily happen that you set the attributes and this is against the task (see [here](https://zulip.in.tum.de/#narrow/stream/1525-PGdP-W11H03/topic/login.20Methode.20Attribute/near/909609))

**Comment by felixhauptmann** [comment]
I kind of agree with you... Even though Konrad stated that [it is irrelevant](https://zulip.in.tum.de/#narrow/stream/1525-PGdP-W11H03/topic/login.20Methode.20Attribute/near/910480) we can close this because enforcing the behavior doesn't hurt.

**Pull request closed without merging**